knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java

```diff
@@ -51,28 +51,34 @@ static KnotxAssemblerHandler create(Vertx vertx, KnotxServerConfiguration config
   public void handle(RoutingContext context) {
     KnotContext knotContext = context.get("knotContext");
 
-    assembler.processObservable(knotContext)
-        .doOnNext(this::traceMessage)
-        .subscribe(
-            ctx -> {
-              if (ctx.getClientResponse().getStatusCode() == HttpResponseStatus.OK.code()) {
-                sendResponse(context, ctx);
-              } else {
-                context.fail(ctx.getClientResponse().getStatusCode());
+    if (knotContext.getClientResponse().getStatusCode() != HttpResponseStatus.OK.code()) {
+      HttpServerResponse httpResponse = context.response();
```

Comment:
Why not sendResponse() method here? We can slightly modify this method to support both cases.